core/controllers/learner_group.py

```diff
@@ -827,3 +958,135 @@ def get(self, username):
+        learner_groups_joined = (
+            learner_group_fetchers.get_learner_groups_joined_by_learner(
+                self.user_id)
+        )
+        learner_of_learner_groups_data = []
+        for learner_group in learner_groups_joined:
+            learner_of_learner_groups_data.append({
+                'id': learner_group.group_id,
+                'title': learner_group.title,
+                'description': learner_group.description,
+                'facilitator_usernames': user_services.get_usernames(
+                    learner_group.facilitator_user_ids),
+                'learners_count': len(learner_group.learner_user_ids)
+            })
+
+        self.render_json({
+            'learner_groups_joined': learner_of_learner_groups_data,
+            'invited_to_learner_groups': invited_to_learner_groups_data
+        })
+
+
+class LearnerGroupProgressSharingPermissionHandler(base.BaseHandler):
+    """The handler for fetching and updating progress sharing permissions of
+    a learner for a given learner group.
+    """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    URL_PATH_ARGS_SCHEMAS = {
+        'learner_group_id': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'is_regex_matched',
+                    'regex_pattern': constants.LEARNER_GROUP_ID_REGEX
+                }]
+            },
+            'default_value': None
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'PUT': {
+            'progress_sharing_permission': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': 'false'
+            }
+        }
+    }
+
+    @acl_decorators.can_access_learner_groups
+    def get(self, learner_group_id):
+        """Handles GET requests."""
+
+        progress_sharing_permission = (
+            learner_group_fetchers.can_multi_learners_share_progress(
+                [self.user_id], learner_group_id)[0]
+        )
+        self.render_json({
+            'progress_sharing_permission': progress_sharing_permission
+        })
+
+    @acl_decorators.can_access_learner_groups
+    def put(self, learner_group_id):
+        """Handles PUT requests."""
+
+        progress_sharing_permission = (
+            self.normalized_payload.get(
+                'progress_sharing_permission') == 'true')
+
+        learner_group_services.update_progress_sharing_permission(
+            self.user_id, learner_group_id, progress_sharing_permission)
+
+        self.render_json({
+            'progress_sharing_permission': progress_sharing_permission
+        })
+
+
+class LearnerGroupsFeatureStatusHandler(base.BaseHandler):
+    """The handler for checking whether the learner groups feature is
+    enabled.
+    """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
```

Comment:
Missing types for these constants, please add them as suggested in one of the above comments!

Reply:
Added